packages/world/src/modules/snapsync/SnapSyncSystem.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import { System } from "../../System.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { SliceLib, Slice } from "@latticexyz/store/src/Slice.sol";
+import { DecodeSlice } from "@latticexyz/store/src/tightcoder/DecodeSlice.sol";
+
+import { getKeysInTable } from "../keysintable/getKeysInTable.sol";
+import { KeysInTable, KeysInTableTableId } from "../keysintable/tables/KeysInTable.sol";
+import { SyncRecord } from "./SyncRecord.sol";
+
+function singletonKey(bytes32 key) pure returns (bytes32[] memory) {
+  bytes32[] memory keyTuple = new bytes32[](1);
+  keyTuple[0] = key;
+  return keyTuple;
+}
```

Comment:
"singleton" is a really overloaded term, maybe we should just call it `keyToTuple` or something?

Reply:
This is similar logic to https://github.com/latticexyz/mud/blob/main/packages/world/src/modules/keysintable/query.sol#L23